api-participant/src/main/java/bio/terra/pearl/api/participant/service/CurrentUnauthedUserService.java

```diff
@@ -24,21 +27,22 @@ public CurrentUnauthedUserService(
   }
 
   @Transactional
-  public Optional<CurrentUserService.UserWithEnrollees> unauthedRefresh(
+  public CurrentUserService.UserWithEnrollees unauthedRefresh(
       String token, String portalShortcode, EnvironmentName environmentName) {
     return currentUserService.refresh(token, portalShortcode, environmentName);
   }
 
   @Transactional
-  public Optional<CurrentUserService.UserWithEnrollees> unauthedLogin(
+  public CurrentUserService.UserWithEnrollees unauthedLogin(
       String username, String portalShortcode, EnvironmentName environmentName) {
     Optional<ParticipantUser> userOpt = participantUserDao.findOne(username, environmentName);
-    if (userOpt.isPresent()) {
-      ParticipantUser user = userOpt.get();
-      user = updateUnauthedUserToken(user);
-      return Optional.of(currentUserService.loadFromUser(user, portalShortcode));
+    if (userOpt.isEmpty()) {
+      log.info("User {} not found for environment {}", username, environmentName);
```

Comment:
This logging is helpful since we otherwise do not record this info (and the detail is not exposed in the front end), but note that we are exposing the participant username. Just want to be sure we are deliberate about our PII boundaries, since there will be other cases where info like username will be important for troubleshooting.

Reply:
Yeah, our current marching orders from AppSec are to not log usernames, especially participant usernames.  You're right though, that we don't have any other identifiers at this point.  Maybe we obfuscate the email by just logging first letter + domain?  So e.g. dbush@broadinstitute.org would be `d...@broadinstitute.org`?

Reply:
Thx for clarifying our position on logging PII. Also, I appreciate your workaround suggestion.

I chose to just remove the PII in the couple cases where there was no fallback, and substituted the user ID in the case where it is available. 

I don't think obfuscated username buys us much in these cases, since we have no Juniper data to compare it to (like a list of users in a DB). For cases like this, rare but interesting, we might be better served by logging it to a table, which we can manage with time-to-live records. But for that we should probably wait until we see a specific pattern we are interested in.

Does that make sense?

Reply:
works for me.  the obfuscated username could help since this is the case where they've successfully logged in, and so we'll have record in the B2C access log.  So then we could just match on timestamp and confirm with the obfuscated username.  But I'm not sure that buys us much, so I agree to keep it simple for now and just not log it

Reply:
Ah, great point about the B2C access log.
Hopefully we can match by timestamp for now, and embellish as needed.